+bisos/+iteration/+step/termination.m

```diff
@@ -0,0 +1,134 @@
+classdef termination < bisos.iteration.Step
+   
+properties
+    type = 'termination';
+    varout = [];
+    varin;
+
+    operator;
+    
+    % Structures with info about condition 
+    prev = struct('fhan', @(x) x, 'vars', []);
+    curr = struct('fhan', @(x) x, 'vars', []);
+end
+
+methods
+    function step = termination(~, previous, current, op, varargin)
+        % setup termination rule
+        
+        step.curr = step.setup2struct(step.curr, current);
+        step.varin = step.curr.vars;
```

Comment:
I do think the variable inputs to `step.prev` should be listed in `varin`

```suggestion
        step.varin = union(step.curr.vars, step.prev.vars);
```

Reply:
I'm not sure if that's a good idea, because in a single cycle this step should only run once the `varin` for the current is updated because it's assumed that the `varin` for the previous has been calculated in the previous cycle. But either way, doing that won't change the outcome, only change the place where the step is going to run.

Reply:
In iteration `k`, you want to compare the current (updated) solution to the previous solution; however, since the previous solution is actually pre-stored at iteration `k-1`, the stop should only be run once the variables of `step.prev` have been updated -- otherwise the stored values would represent the solution of `k-2`. Do I talk sense?

Reply:
Understood, due the stored value that's necessary